src/microbe_stage/Spawners.cs

```diff
@@ -222,14 +222,11 @@ public static EntityRecord SpawnChunkWithoutFinalizing(IWorldSimulation worldSim
         entity.Set(new WorldPosition(location,
             new Quaternion(rotationAxis.Normalized(), 2 * Mathf.Pi * (float)random.NextDouble())));
 
-        // TODO: redo chunk visuals with the loadable visual definitions
-        // entity.Set(new PredefinedVisuals
-        // {
-        //     VisualIdentifier = VisualResourceIdentifier.AgentProjectile,
-        // });
-        entity.Set(new PathLoadedSceneVisuals
+        var meshes = chunkType.Meshes;
+
+        entity.Set(new PredefinedVisuals
         {
-            ScenePath = selectedMesh.ScenePath,
+            VisualIdentifier = meshes[random.Next(meshes.Count)].VisualResource,
```

Comment:
This should use the selected mesh because otherwise the visual mesh may not match what was selected as the physics mesh.